File: business_logic/src/test/java/org/metplus/curriculum/domain/job/CreateJobResultHandlerSpy.java
package org.metplus.curriculum.domain.job;

public class CreateJobResultHandlerSpy implements CreateJob.ResultHandler<Void> {
    public String jobExistsWasCalledWith = null;
    public boolean successWasCalled = false;
    public boolean fatalErrorWasCalled = false;

    @Override
    public Void jobExists(String jobId) {
        jobExistsWasCalledWith = jobId;
        return null;
    }

    @Override
    public Void fatalError() {
        fatalErrorWasCalled = true;
        return null;
    }

    @Override
    public Void success() {
        successWasCalled = true;
        return null;
    }
}

File: business_logic/src/test/java/org/metplus/curriculum/domain/job/testDoubles/CreateJobResultHandlerSpy.kt
package org.metplus.curriculum.domain.job.testDoubles

import org.metplus.curriculum.domain.job.CreateJob

class CreateJobResultHandlerSpy : CreateJob.ResultHandler<Void?> {
    var jobExistsWasCalledWith: String? = null
    var successWasCalled = false
    var fatalErrorWasCalledWith: String? = null

    override fun jobExists(jobId: String): Void? {
        jobExistsWasCalledWith = jobId
        return null
    }

    override fun fatalError(jobId: String): Void? {
        fatalErrorWasCalledWith = jobId
        return null
    }

    override fun success(): Void? {
        successWasCalled = true
        return null
    }
}
